### ISSUE TITLE
bug: dead code (?) throwing an error at import time

### ISSUE THREAD

**Issue by jettdc** [ibis-project/ibis-substrait#844]
https://github.com/ibis-project/ibis-substrait/blob/2edf1fbc8cd2610fbd341eeaa2013c384080a578/ibis_substrait/compiler/translate.py#L1333-L1350

Looks like this code is unused, but since `ops.NotExistsSubquery` doesn't exist, it throws.

```
AttributeError: module 'ibis.expr.operations' has no attribute 'NotExistsSubquery'
```

Curious as to how this code got through CI and was pushed to prod as well

**Comment by cpcloud** [comment]
Thanks for the issue @jettdc!

We aren't testing against the latest version of `ibis-framework` in CI.

We've been prioritizing other things in the meantime.

Just so we can figure out how to prioritize any ibis-substrait work, what's your level of interest in `ibis-substrait`? Can you describe a bit about what you're doing with it?

**Comment by jettdc** [comment]
Personally I'm mostly interested in the substrait work (over the ibis direct connections ). For my usecases it's most valuable to just have a generic dataframe manipulation library decoupled from the execution, especially for usecases where you might want to hot swap a backend after having already generated a substrait plan from the ibis expression. This is useful where, for example, sometimes you want to run the same code in memory vs in a distributed environment without rewriting the code.

**Comment by jettdc** [comment]
I think where Ibis is especially powerful is in it's ability to define a logical plan once in a dead simple way and then use that forever in a variety of ways. The interactive stuff and whatnot that I see a lot of in the ibis docs is not especially useful for me

**Comment by lostmygithubaccount** [comment]
> For my usecases it's most valuable to just have a generic dataframe manipulation library decoupled from the execution

this is very inline with the portability aspect of Ibis. however, most backends (data platforms) today don't support substrait consumption, so there's a bit of a chicken-and-egg problem in making this a reality. ideally, we'd love to generate and send Substrait for most backends instead of SQL

> The interactive stuff and whatnot that I see a lot of in the ibis docs is not especially useful for me

this is interesting -- production code (non-interactive) is definitely important, but in my experience data engineers/analysts/scientists tend to build up their queries interactively and inspect outputs as they go. how are you validating your queries as you're building them?

**Comment by jettdc** [comment]
Yep totally get substrait being early and not being supported in a lot of places yet, though I really do think it's adoption will pick up relatively soon

> in my experience data engineers/analysts/scientists tend to build up their queries interactively and inspect outputs as they go  
I'll agree with you that data analysts and scientists tend to prefer and use interactive forms of data exploration, but I disagree that serious data engineers do the same (outside of data exploration). From what I've seen, in the vast majority of cases of building production data pipelines with tools like spark, the code is written and run on triggered builds, not interactively. And for debugging, there are various tools for debugging code like this, even as simple as just previewing your output at each step that you write, or unit testing your code.

**Comment by gforsyth** [comment]
Resolved via #853
